lib/Doctrine/Migrations/Exception/MigrationConfigurationConflict.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Exception;
+
+use Doctrine\Migrations\AbstractMigration;
+use UnexpectedValueException;
+
+use function get_class;
+use function sprintf;
+
+final class MigrationConfigurationConflict extends UnexpectedValueException implements MigrationException
+{
+    public static function migrationIsNotTransactional(AbstractMigration $migration): self
+    {
+        return new self(
+            sprintf(
+                "Context: attempting to execute migrations with all-or-nothing enabled\n" .
+                "Problem: migration %s is marked as non-transactional\n" .
+                'Solution: disable all-or-nothing in configuration or by command-line option, or enable transactions for all migrations',
+                get_class($migration)
+            )
+        );
```

Comment:
Let's use linebreaks only when necessary, and let's use heredoc to avoid having to concatenated 3 strings and manage line returns.
```suggestion
        return new self(sprintf(
            <<<'EXCEPTION'
Context: attempting to execute migrations with all-or-nothing enabled
Problem: migration %s is marked as non-transactional
Solution: disable all-or-nothing in configuration or by command-line option, or enable transactions for all migrations
EXCEPTION,
            get_class($migration)
        ));
```

Reply:
I didn't want to use heredoc because the PHP version requirement is >=7.2, which means we can't indent the heredoc, which is incredibly ugly to me. 🤷‍♂️